client/src/components/admin/FlagsPopup.tsx

```diff
@@ -25,22 +26,26 @@ const FlagsPopup = ({ close }: FlagsPopupProps) => {
     }, []);
 
     useEffect(() => {
+        const flagsCopy = [...flags];
```

Comment:
oops i did call sort on an immutable thing but u should prob use this function instead of manually copying the list:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted